model/transfer.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/jinzhu/gorm/dialects/postgres"
+)
+
+type (
+	// TransferDestination schema for transfer destination
+	TransferDestination struct {
+		Type            string          `json:"type"`
+		BankDetails     BankDetails     `json:"bank_details"`
+		PersonalDetails PersonalDetails `json:"personal_details"`
+	}
+
+	// InitiateTransferRequest schema for initiate transfer request
+	InitiateTransferRequest struct {
+		CustomerID  string              `json:"customer_id"`
+		Amount      float64             `json:"amount"`
+		Currency    string              `json:"currency"`
+		Destination TransferDestination `json:"destination"`
+		Note        string              `json:"note,omitempty"`
+		Reason      string              `json:"reason"`
+		Reference   string              `json:"reference"`
+	}
+
+	// InitiateTerminalTransferRequest schema for initiate terminal transfer request
+	InitiateTerminalTransferRequest struct {
+		Amount              float64              `json:"amount"`
+		SourceCurrency      string               `json:"source_currency"`
+		DestinationCurrency string               `json:"destination_currency"`
+		UseBalance          string               `json:"use_balance"`
+		BeneficiaryID       *string              `json:"beneficiary_id,omitempty"`
+		Destination         *TransferDestination `json:"destination,omitempty"`
+		Note                *string              `json:"note,omitempty"`
+		Reason              string               `json:"reason"`
+	}
+
+	// TransferResponse schema for transfer response
+	TransferResponse struct {
+		ID uuid.UUID `json:"id"`
+		InitiateTransferRequest
+		CreatedAt time.Time `json:"created_at"`
+		Status    string    `json:"status"`
+	}
+
+	// ExchangeRateDetails schema for exchange rate details
+	ExchangeRateDetails struct {
+		ExchangeRate     float64 `json:"exchange_rate"`
+		FeeFlat          float64 `json:"flat_fee"`
+		FeePercentage    float64 `json:"fee_percentage"`
+		FeeAmount        float64 `json:"fee_amount"`
+		AmountReceivable float64 `json:"amount_receivable"`
+	}
+
+	// Transfer schema for customer transfer
+	Transfer struct {
+		ID              uuid.UUID      `json:"id"`
+		Name            string         `json:"name"`
+		Email           string         `json:"email"`
+		CustomerID      uuid.UUID      `json:"customer_id"`
+		Amount          float64        `json:"amount"`
+		Currency        string         `json:"currency"`
+		Destination     postgres.Jsonb `json:"destination"`
+		Note            *string        `json:"note"`
+		Reason          string         `json:"reason"`
+		CreatedAt       time.Time      `json:"created_at"`
+		CompletedAt     sql.NullTime   `json:"completed_at"`
+		UpdatedAt       sql.NullTime   `json:"updated_at"`
+		BatchDate       time.Time      `json:"batch_date"`
+		Status          string         `json:"status"`
+		Reference       string         `json:"reference"`
+		CancelReason    *string        `json:"cancel_reason"`
+		TransactionType string         `json:"type"`
```

Comment:
there's a mismatch btw the json tag and field name

Reply:
`type` is being used as the query param on external api

Reply:
noted